packages/gestalt-datepicker/src/DateRange/InternalDatePicker.tsx

```diff
@@ -73,9 +111,17 @@ const InternalDatePickerWithForwardRef = forwardRef<HTMLInputElement, ModifiedPr
             calendarClassName={styles['react-datepicker-inline']}
             dateFormat={format}
             dayClassName={() => styles['react-datepicker__days']}
-            endDate={rangeEndDate ?? undefined}
+            endDate={
+              selectedRange === DateRangeType.Primary
+                ? rangeEndDate ?? undefined
+                : secondaryRangeEndDate ?? undefined
+            }
             excludeDates={excludeDates && [...excludeDates]}
-            highlightDates={initRangeHighlight ? [initRangeHighlight] : []}
+            highlightDates={
+              selectedRange === DateRangeType.Primary
+                ? generateHighligths(secondaryRangeStartDate, secondaryRangeEndDate)
```

Comment:
Ah so you need the inverse of the selection for the highlights?

Reply:
Yes, here we want to highlight the "inactive" range, so, if the "active" range is the `primary`, then we want to highlight the secondary, and vice versa.